apps/examples/st_things/msghandling/sample_msghandling_things.c

```diff
@@ -0,0 +1,158 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *	  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <signal.h>
+#include <st_things/st_things.h>
+#include <iotbus/iotbus_gpio.h>
+
+#define TAG		"LIGHT_THINGS"
+
+static const char *g_res_switch = "/capability/switch/main/0";
+
+static const char* g_res_color_temperature_res_uri = "/capability/colorTemperature/main/0";
+static const char* g_res_door_controluri = "/capability/doorControl/main/0";
+static const char* g_res_airquality_res_uri = "/capability/airQualitySensor/main/0";
+static const char* g_res_tone_uri = "/capability/tone/main/0";
+static const char* RES_WWSTFILTERSTATERESURI = "/capability/filterStatus/main/0";
+
+extern bool handle_get_request_on_switch(st_things_get_request_message_s *req_msg, st_things_representation_s *resp_rep);
+extern bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep);
+
+extern bool handle_get_request_on_resource_wwstcolortemperatureresuri(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_set_request_on_resource_wwstcolortemperatureresuri(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_get_request_on_resource_doorcontrol(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_set_request_on_resource_doorcontrol(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_get_request_on_resource_wwstairqualitysensorlevelresuri(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_get_request_on_resource_tone(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_set_request_on_resource_tone(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep);
+
+extern bool handle_get_request_on_resource_wwstfilterstateresuri(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_set_request_on_resource_wwstfilterstateresuri(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep);
+
+#ifdef CONFIG_RESET_BUTTON
+static bool check_reset_button_pin_number(void)
```

Comment:
CONFIG_RESET_BUTTON, "reset" can be commonly used on each module,
would you please consider to provide different feauture name?